app/services.go

```diff
@@ -68,15 +77,82 @@ type serviceManager struct {
 	err    error
 }
 
+type PushServiceCert struct {
+	*x509.Certificate
+	PrivateKey interface{}
+}
+
+func (s *serviceManager) loadPushCerts() {
+	if s.err != nil {
+		return
+	}
+	if s.APNS.PrivateKeyPath == "" {
+		pkcs12Data, err := ioutil.ReadFile(s.APNS.CertificatePath)
+		if err != nil {
+			s.err = err
+			return
+		}
+		s.PushServiceCert.PrivateKey, s.PushServiceCert.Certificate, err =
+			pkcs12.Decode(pkcs12Data, s.APNS.PrivateKeyPass)
+		if err != nil {
+			s.err = err
+			return
+		}
+	} else {
```

Comment:
up to you, but I'd clean it up a bit by doing the following:

```
s.PushServiceCert.PrivateKey, s.PushServiceCert.Certificate, s.err = pkcs12.Decode(pkcs12Data, s.APNS.PrivateKeyPass)
return
```
the drop the `else`(since we return right above), and then do the same on line 123.